### PR TITLE
[MAINT] -  add pre-commit to GHA CI pipeline

### PR DESCRIPTION
**Pull request by HarshCasper**
Fixes https://github.com/nebari-dev/nebari/issues/66

### CODE REVIEW

File: docs/docs/tutorials/creating-cds-dashboard.md [hunk 1]
```diff
@@ -1,20 +1,17 @@
...
 
 ## What is a dashboard?
 
 Analyzing data provides us with insights. A dashboard stitches these insights together into a meaningful story.
 
-Different tools and technologies can be used to create and deploy a dashboard. In this tutorial
-we will be building a simple dashboard on Nebari which can be shared with other users and groups via
-[ContainDS Dashboards](https://cdsdashboards.readthedocs.io/en/stable/) (CDS Dashboards).
+Different tools and technologies can be used to create and deploy a dashboard. In this tutorial we will be building a simple dashboard on Nebari which can be shared with other
+users and groups via [ContainDS Dashboards](https://cdsdashboards.readthedocs.io/en/stable/) (CDS Dashboards).
```
Comment (kcpevey) [inline docs/docs/tutorials/creating-cds-dashboard.md:12]: which hook is unwrapping these lines? That's very frustrating as a writer. mdformat is set to 180. That needs to be changed to 120. That may be expanding to 120 in addition to cutting things off at 120.

File: docs/docs/quickstart.md [hunk 2]
```diff
@@ -1,5 +1,5 @@
----
-id: quickstart
----
+______________________________________________________________________
+
+## id: quickstart
```
Comment (kcpevey) [inline docs/docs/quickstart.md:3]: What caused this? its not correct

**Review by kcpevey — CHANGES_REQUESTED** [review]

File: docs/docs/quickstart.md (hunk 2 above)
Comment (trallard) [reply docs/docs/quickstart.md:3]: Oh boy it's the https://mdformat.readthedocs.io/en/stable/users/style.html one - it only uses ATX headings and do not recognise frontmatter - also this one overwrites the 180 lines

Can we remove this pre-commit? it is going to cause a whole lot of issues.

Comment (HarshCasper) [reply docs/docs/quickstart.md:3]: Sure. I will do it.

**Pull request closed without merging**